### PR TITLE
refactor: (GenAI) Reorganized Function calling Samples (Group A) 

### PR DESCRIPTION
**Pull request by Thoughtseize1**
#### Created new folder `/generative_ai/function_calling/`
| Region_tag                                                   | Old file                                                | New file                   |
|--------------------------------|----------------------------------|----------------|
| generativeaionvertexai_gemini_function_calling_chat                    | function_calling_chat.py                                     | chat_example.py                 |
| generativeaionvertexai_gemini_function_calling_advanced                | function_calling.py                                          | advanced_example.py             |
| generativeaionvertexai_gemini_function_calling                         | function_calling.py                                          | basic_example.py                |
| generativeaionvertexai_gemini_chat_completions_function_calling_basic  | chat_completions/chat_completions_function_calling_basic.py  | chat_function_calling_basic.py  |
| generativeaionvertexai_gemini_chat_completions_function_calling_config | chat_completions/chat_completions_function_calling_config.py | chat_function_calling_config.py |

**❗I figured out an issue with `generativeaionvertexai_gemini_function_calling_advanced`. 
A detailed description is presented in the [document](https://docs.google.com/document/d/1kt2ucN36TkpQOsfy5G1hR31m9KzG8WGa8VgpgA7zTwY/edit?usp=sharing).**
- --
**❗Files need to be deleted after approval:**

- /generative_ai/function_calling_chat.py
- /generative_ai/function_calling_chat_test.py
- /generative_ai/function_calling.py
- /generative_ai/function_calling_test.py
- /generative_ai/chat_completions/chat_completions_function_calling_basic.py
- /generative_ai/chat_completions/chat_completions_function_calling_config.py
- /generative_ai/chat_completions/chat_completions_test.py (**only marked 2 tests**)
- --
## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: generative_ai/function_calling/advanced_example.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    # [START generativeaionvertexai_gemini_function_calling_advanced]
+    import vertexai
+    from vertexai.preview.generative_models import (
+        FunctionDeclaration,
+        GenerativeModel,
+        Tool,
+        ToolConfig,
+    )
+
+    # TODO (developer): Uncomment and replace line with your project ID
```
Comment (msampathkumar) [inline generative_ai/function_calling/advanced_example.py:32]:
```suggestion
    # TODO(developer): Update & uncomment below line
```

File: generative_ai/function_calling/basic_example.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    from vertexai.generative_models import (
+        Content,
+        FunctionDeclaration,
+        GenerationConfig,
+        GenerativeModel,
+        Part,
+        Tool,
+    )
+
+    # TODO (developer): Uncomment and replace line with your project ID
```
Comment (msampathkumar) [inline generative_ai/function_calling/basic_example.py:34]:
```suggestion
    # TODO(developer): Update & uncomment below line
```

File: generative_ai/function_calling/chat_example.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    import vertexai
+    from vertexai.generative_models import (
+        FunctionDeclaration,
+        GenerationConfig,
+        GenerativeModel,
+        Part,
+        Tool,
+    )
+
+    # Initialize Vertex AI
```
Comment (msampathkumar) [inline generative_ai/function_calling/chat_example.py:32]:
```suggestion
```

File: generative_ai/function_calling/chat_example.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    # TODO (developer): Uncomment and set the PROJECT_ID variable
```
Comment (msampathkumar) [inline generative_ai/function_calling/chat_example.py:33]:
```suggestion
    # TODO(developer): Update & uncomment below line
```

File: generative_ai/function_calling/advanced_example.py (hunk 1 above)
Comment (Thoughtseize1) [reply generative_ai/function_calling/advanced_example.py:32]: Done

File: generative_ai/function_calling/chat_example.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    # PROJECT_ID = "your-project-id"
+    vertexai.init(project=PROJECT_ID, location="us-central1")
```
[the remark this answers is not in the document]
Comment (Thoughtseize1) [reply generative_ai/function_calling/chat_example.py:35]: Done

File: generative_ai/function_calling/chat_example.py (hunk 3 above)
Comment (Thoughtseize1) [reply generative_ai/function_calling/chat_example.py:32]: Done

File: generative_ai/function_calling/chat_example.py (hunk 4 above)
Comment (Thoughtseize1) [reply generative_ai/function_calling/chat_example.py:33]: Done

File: generative_ai/function_calling/chat_function_calling_basic.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+
+def generate_text() -> object:
+    # [START generativeaionvertexai_gemini_chat_completions_function_calling_basic]
+    import vertexai
+    import openai
+
+    from google.auth import default, transport
+
+    # TODO (developer): Uncomment and replace line with your project ID
```
Comment (msampathkumar) [inline generative_ai/function_calling/chat_function_calling_basic.py:27]:
```suggestion
    # TODO(developer): Update & uncomment below line
```

File: generative_ai/function_calling/chat_function_calling_config.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+
+def generate_text() -> object:
+    # [START generativeaionvertexai_gemini_chat_completions_function_calling_config]
+    import vertexai
+    import openai
+
+    from google.auth import default, transport
+
+    # TODO (developer): Uncomment and replace line with your project ID
```
Comment (msampathkumar) [inline generative_ai/function_calling/chat_function_calling_config.py:27]:
```suggestion
    # TODO(developer): Update & uncomment below line
```

Comment (Thoughtseize1) [reply generative_ai/function_calling/chat_function_calling_config.py]: Done

File: generative_ai/function_calling/chat_function_calling_basic.py (hunk 6 above)
Comment (Thoughtseize1) [reply generative_ai/function_calling/chat_function_calling_basic.py]: Done

File: generative_ai/function_calling/advanced_example.py [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    # PROJECT_ID = "your-project-id"
+
+    # Initialize Vertex AI
+    vertexai.init(project=PROJECT_ID, location="us-central1")
+
+    # Specify a function declaration and parameters for an API request
+    get_product_sku_func = FunctionDeclaration(
+        name="get_product_sku",
+        description="Get the available inventory for a Google products, e.g: Pixel phones, Pixel Watches, Google Home etc",
+        # Function parameters are specified in OpenAPI JSON schema format
```
Comment (koverholt) [inline generative_ai/function_calling/advanced_example.py:42]: I was discussing this with some folks on the OpenAPI schema technical steering committee, and they suggested we refer to this as `JSON schema format` and drop the OpenAPI part for technical accuracy. So I suggest making that change here and throughout all Function Calling code samples. And I'm working on getting this updated in the Function Calling documentation as well.

**Review by koverholt — APPROVED** [review]
Thanks for the reorg and cleanup. I left one comment, everything else LGTM. Thanks!

File: generative_ai/function_calling/advanced_example.py (hunk 8 above)
Comment (Thoughtseize1) [reply generative_ai/function_calling/advanced_example.py]: Done

**Review by msampathkumar — APPROVED** [review]
LGTM

**Pull request merged**